src/lib.rs

```diff
@@ -9,25 +9,36 @@
 #![deny(non_snake_case)]
 #![deny(unused_mut)]
 #![deny(missing_docs)]
-
 // Experimental features we need.
 #![cfg_attr(bench, feature(test))]
 #![cfg_attr(docsrs, feature(doc_cfg))]
 
 #[cfg(bench)]
 extern crate test;
 
-use std::cmp::Reverse;
+#[cfg(any(test, feature = "rand"))]
+mod single_random_draw;
 
 #[cfg(any(test, feature = "rand"))]
-use rand::{seq::SliceRandom, thread_rng};
+mod errors;
 
-/// Trait that a UTXO struct must implement to be used as part of the coin selection
-/// algorithm.
-pub trait Utxo: Clone {
-    /// Return the value of the UTXO.
-    fn get_value(&self) -> u64;
-}
+#[cfg(any(test, feature = "rand"))]
+use rand::thread_rng;
+
+#[cfg(any(test, feature = "rand"))]
```

Comment:
instead of having to put this everywhere, you can force the `rand` feature to be used in tests by putting this under the dev-dependencies in the cargo file:
```
rust-bitcoin-coin-selection = {path = ".", features = ["rand"]}
```

Reply:
Thanks!  Good idea.

Reply:
removed all feature gates and made rand default for dev mode